### PR TITLE
feat: adds page header component

### LINKED ISSUE #6437: Add new mobile navigation elements
### Overview

As part of this rework we will need to change the way the mobile nav bar works.

We will introduce a centered item, with the trade icon. That will pop open an action menu where users can:

1. Send
2. Receive
3. Trade
4. Scan QR Code
5. Buy

In addition to that, we will also need to show or hide this nav on sub pages. And also add a navigation header on top of these sub pages.

Also might be worth looking into the possibility of allowing users to swipe on mobile to go back. 

https://www.npmjs.com/package/react-swipeable-routes

### References and additional details

MobileNavBar, and the Layout

### Acceptance Criteria

User can tap the blue center button, and access any of the main features of the app easily.

When on sub pages, they take up the entire screen and the mobile nav bar is hidden. They can also navigate back using the back arrow to the other pages.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### LINKED ISSUE #6496: Mobile Sub header
### Overview

Add a new mobile header for sub pages for that more native feel

### References and additional details

This will be a new component

### Acceptance Criteria

On Sub pages in the mobile app, there should be a header that has a back arrow.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

Adds a highly composable header component to be used for desktop and mobile.

Will require a few follow up PRs to plug this in in all the places.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #6496  

Low risk

> What protocols, transaction types or contract interactions might be affected by this PR?

### Operations

Right now the only view that has this is the asset view. So if you tap into an asset from the dashboard you should see a new header at the top with a back arrow and as you scroll down the title will be replaced with the price.

Desktop should look the same as it does now.

## Screenshots (if applicable)
Simulator Screenshot - iPhone 15 Pro Max - 2024-03-19 at 12 24 25
Simulator Screenshot - iPhone 15 Pro Max - 2024-03-19 at 12 24 28

### CODE REVIEW

**Review by 0xApotheosis — COMMENTED** [review]
My first observation on functional testing - is it meant to end up in a weird half-way state for medium width screens (between 768-1114, it seems)?

https://github.com/shapeshift/web/assets/97164662/c67de72c-d41d-419c-9c1b-64b06495e0b2

**Comment by reallybeard** [comment]
@0xApotheosis 

> My first observation on functional testing - is it meant to end up in a weird half-way state for medium width screens (between 768-1114, it seems)?
> 
>  page-header.mp4

I'll take a look

File: src/components/ScrollDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  threshold?: number // Threshold for when to switch displays, default to 72px
+} & PropsWithChildren
+
+type ScrollDisplayCompoundProps = {
+  Default: React.FC<{ children: React.ReactNode }>
+  OutOfView: React.FC<{ children: React.ReactNode }>
+}
+
+export const ScrollDisplay: React.FC<ScrollDisplayProps> & ScrollDisplayCompoundProps = ({
+  threshold = 72,
```
Comment (0xApotheosis) [inline src/components/ScrollDisplay.tsx:17]: We should move this guy to a constant to avoid the magic number.

**Review by 0xApotheosis — COMMENTED** [review]
Looks sane, just the comment above re: the medium width.

Non-blocking magic number request.

File: src/components/ScrollDisplay.tsx (hunk 1 above)
Comment (reallybeard) [reply src/components/ScrollDisplay.tsx:17]: @0xApotheosis is there a constant file I can use, or should I put one somewhere else?

Comment (0xApotheosis) [reply src/components/ScrollDisplay.tsx]: Same file is fine for single use 🤝

**Review by 0xApotheosis — APPROVED** [review]
There we go!

**Pull request merged**